test/test_jit.py

```diff
@@ -15182,6 +15182,38 @@ def forward(self, x):
         with self.assertRaisesRegex(RuntimeError, 'has no attribute'):
             torch.jit.script(wrapped)
 
+    def test_rescripting_loaded_modules(self):
+        class InnerSubmod(nn.Module):
+            def __init__(self):
+                super().__init__()
+                self.register_buffer("foo", torch.ones(1))
```

Comment:
Consider adding a constant, parameter and regular data attribute to either `InnerSubmod` or `Inner` to make sure those code paths are tested too.